core/store/src/db/rocksdb.rs

```diff
@@ -450,7 +492,14 @@ fn rocksdb_options(store_config: &StoreConfig, mode: Mode) -> Options {
         opts.increase_parallelism(std::cmp::max(1, num_cpus::get() as i32 / 2));
         opts.set_max_total_wal_size(bytesize::GIB);
     }
+    opts
+}
 
+fn rocksdb_options(store_config: &StoreConfig, mode: Mode) -> Options {
+    let mut opts = common_rocksdb_options();
+    opts.create_missing_column_families(mode.read_write());
+    opts.create_if_missing(mode.can_create());
+    opts.set_max_open_files(store_config.max_open_files.try_into().unwrap_or(i32::MAX));
```

Comment:
Not related but this default could be disastrous. The higher the limit the fewer DBs we can open.